### PR TITLE
Move unit tests to GitHub workflows, test also under Python 3.9

### PR DESCRIPTION
**Pull request by nsoranzo**
Also some random fixes to other GitHub workflows.

### CODE REVIEW

File: .github/workflows/unit.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Unit tests
+on: [push, pull_request]
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.6']
```
Comment (mvdbeek) [inline .github/workflows/unit.yaml]:
```suggestion
        python-version: ['3.6', '3.9']
```

File: .github/workflows/unit.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Unit tests
+on: [push, pull_request]
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (mvdbeek) [inline .github/workflows/unit.yaml:9]:
```suggestion
      fail-fast: false
      matrix:
```

**Review by mvdbeek — APPROVED** [review]
Should we test more python versions ?

File: .github/workflows/unit.yaml (hunk 1 above)
Comment (nsoranzo) [reply .github/workflows/unit.yaml]: Maybe 3.6 and 3.9, like for lint workflow?

**Comment by dannon** [comment]
Unit test failure looks real -- fractions.gcd import failing in 3.9?  (math.gcd now?)

**Comment by nsoranzo** [comment]
> Unit test failure looks real -- `fractions.gcd` import failing in 3.9? (math.gcd now?)

Yes, unfortunately it's not in our code but in `networkx`, which comes in (pinned to 1.11) as a depedency of [isa-rwval](https://github.com/ISA-tools/isa-rwval). The `fractions.gcd` import is fixed in networkx >= 2.4 .

Not sure what's the best course of action here, any suggestion @djcomlab ?

**Comment by djcomlab** [comment]
> depedency of [isa-rwval](https://github.com/ISA-tools/isa-rwval). The `fractions.gcd` import is fixed in networkx >= 2.4 .
> 
> Not sure what's the best course of action here, any suggestion @djcomlab ?

I believe `isa-rwval` can probably be updated quickly. I will check it today and let you know.

**Comment by nsoranzo** [comment]
> Should we test more Python versions ?

@mvdbeek I was thinking of adding '3.9' also to `.github/workflows/first_startup.yaml`, what do you think?

**Comment by mvdbeek** [comment]
Sounds good

**Comment by nsoranzo** [comment]
All tests pass now! Thanks @djcomlab for the help!

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**